src/Hardmon.cpp

```diff
@@ -0,0 +1,122 @@
+#include <Hardmon.hpp>
+
+namespace VCU {
+
+Hardmon::Hardmon(reqGPIO gpios, IO::CAN& ptCAN) : gpios(gpios), powertrainCAN(ptCAN) {
+}
+
+CO_OBJ_T* Hardmon::getObjectDictionary() {
+    return &objectDictionary[0];
+}
+
+uint8_t Hardmon::getNumElements() {
+    return OBJECT_DICTIONARY_SIZE;
+}
+
+uint8_t Hardmon::getNodeID() {
+    return NODE_ID;
+}
+
+void Hardmon::handlePowertrainCanMessage(IO::CANMessage& message) {
+    switch (message.getId()) {
+    case DEV::PowertrainCAN::HIB_MESSAGE_ID:
+        forwardEnable = powertrainCAN.parseHIBForwardEnable(message);
+        break;
+    case DEV::PowertrainCAN::UC_SELF_TEST_MESSAGE_ID:
+        powertrainCAN.sendHardmonSelfTestResponse();
+        break;
+    default:
+        //we don't care about this message lol
+        break;
+    }
+}
+
+EVT::core::types::FixedQueue<POWERTRAIN_QUEUE_SIZE, IO::CANMessage>* Hardmon::getPowertrainQueue() {
+    return &powertrainCAN.queue;
+}
+
+void Hardmon::process() {
+    //handle all the powertrain CAN messages
+    IO::CANMessage message;
+    while (!powertrainCAN.queue.isEmpty()) {
```

Comment:
Don't change this because it will change with the RTOS anyway, but be very careful with stuff like this in general. If the bus was very busy, you're liable to get stuck in this loop for a long time and have major problems